particles/Tutorial/Kotlin/Demo/TTTGame.kt

```diff
@@ -0,0 +1,173 @@
+            p1.id = 0.0
+            playerOne.set(p1)
+        }
+        if (handle.name.equals("playerTwo") && playerTwo.get()?.id != 1.0) {
+            val p2 = playerTwo.get() ?: TTTGame_PlayerOne()
+            p2.id = 1.0
+            playerTwo.set(p2)
+        }
+    }
+
+    override fun populateModel(slotName: String, model: Map<String, Any?>): Map<String, Any?> {
+        val gs = gameState.get() ?: TTTGame_GameState()
+
+        val cp = gs.currentPlayer ?: -1.0
+        val p1 = playerOne.get() ?: TTTGame_PlayerOne()
+        val p2 = playerTwo.get() ?: TTTGame_PlayerTwo()
+
+        val cpName = if (cp == p1.id) p1.name else p2.name
+        val cpAvatar = if (cp == p1.id) p1.avatar else p2.avatar
+
+        val winnerAvatar = gs.winnerAvatar ?: ""
+        val winnerName = if (winnerAvatar == p1.avatar) p1.name else p2.name
+        val congratsMessage = if (gs.winnerAvatar != null)
+            "Congratulations $winnerName, you won!" else "It's a tie!"
+
+        return model + mapOf(
+            "message" to if (gs.gameOver ?: false) congratsMessage else "",
+            "playerDetails" to "$cpName playing as $cpAvatar"
+        )
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        val gs = gameState.get() ?: TTTGame_GameState()
+        // Apply the moves
+        if (!(gs.gameOver ?: false)) {
+            val board = gs.board ?: defaultGame.board!!
+            val boardList = board.split(",").toMutableList()
+            // Check the handle updated matches the current player
+            if (handle.name.equals("playerOneMove") && gs.currentPlayer == 0.0) {
+                applyMove(
+                    mv = playerOneMove.get()?.move?.toInt() ?: -1,
+                    avatar = playerOne.get()?.avatar ?: "",
+                    boardList = boardList,
+                    gs = gs
+                )
+            } else if (handle.name.equals("playerTwoMove") && gs.currentPlayer == 1.0) {
+                applyMove(
+                    mv = playerTwoMove.get()?.move?.toInt() ?: -1,
+                    avatar = playerTwo.get()?.avatar ?: "",
+                    boardList = boardList,
+                    gs = gs
+                )
+            }
+        }
+        if (hasReset()) {
+            gameState.set(defaultGame)
+            playerOneMove.set(TTTGame_PlayerOneMove())
+            playerTwoMove.set(TTTGame_PlayerTwoMove())
+            events.clear()
+        }
+        renderOutput()
+    }
+
+    override fun getTemplate(slotName: String): String {
+
+        return """
+            It is your turn <span>{{playerDetails}}</span>.
+            <div slotid="boardSlot"></div>
+            <div><span>{{message}}</span></div>
+            """.trimIndent()
+    }
+
+    private fun applyMove(
+        mv: Int,
+        avatar: String,
+        boardList: MutableList<String>,
+        gs: TTTGame_GameState
+    ) {
+        if (mv > -1 && mv < 10 && boardList[mv] == "") {
```

Comment:
This is a good situation for an extension function on `Int`:

```kotlin
private fun Int.isValidMove(boardList: List<String>) = this in 0..9 && boardList[this] == ""
```

This'd make it so you could do something like:

```kotlin
private fun applyMove(....) {
    if (!mv.isValidMove()) return
    ...
}